### ISSUE TITLE
opam configuration seems to change accidentally

### ISSUE THREAD

**Issue by MdeLv** [ocaml/opam#4186]
debian 10
ocaml.4.07.1
opam 2.0.7

A few years ago (and still today), I had this problem when following this instruction from `opam init`:
```
$ opam init -a

User configuration:
  ~/.profile is already up-to-date.
[NOTE] Make sure that ~/.profile is well sourced in your ~/.bashrc.
```
If I `source ~/.profile` in ~/.bashrc, then I have a loop and must do C-c to interrupt it.

I use mainly these files:
~/.profile
~/.bashrc
~/.ocamlinit

To get opam working, I have:
```
$ cat ~/.profile    # "RIGHT" version
# opam configuration
test -r /home/dev/.opam/opam-init/init.sh && . /home/dev/.opam/opam-init/init.sh > /dev/null 2> /dev/null || true
```
```
$ cat ~/.bashrc    # RIGHT version
# opam configuration
 test -r /home/dev/.opam/opam-init/init.sh && . /home/dev/.opam/opam-init/init.sh > /dev/null 2> /dev/null || true
```
If I source ~/.profile, then it loops:
```
$ cat ~/.bashrc    # BAD version, making a loop when a new shell is opened 
                   # or ~/.bashrc sourced
source ~/.profile
```

Q1: Is the message "Make sure that ~/.profile is well sourced in your ~/.bashrc" correct?
What should be the exact ~/.bashrc AND ~/.profile setup to get opam working this automatic way when opening a new shell?

Q2: sometimes, opam (auto-)complete is suddenly disabled during usual opam operations.
It's clear that ~/.opam/opam-init/init.sh is supposed to enable it.
It's unclear what really enables or disables it. Can you explain the right way to enable/disable it and to check that it's enabled/disabled?

Thanks.

**Comment by rjbou** [comment]
As said in #3990 _Shell startup loading can be nonsensical sometimes, and quite difficult to raise a general usability consensus (cf. #1447, #533, etc.)._

Opam proposes you to choose which file to update, using `~/profile` as a default.

> Q1: Is the message "Make sure that ~/.profile is well sourced in your ~/.bashrc" correct?

This message is generated using the chosen file. It's a hint in case append file is not sourced.

> Q2: sometimes, opam (auto-)complete is suddenly disabled during usual opam operations.

Is it possible to identify after what operation completion suddenly disable?
If you want to check if completion is still present, you can check that it is present in the `init.sh` file. If you want to be sure to enable/disable some shell configuration, you can re run init with `--{disable,enable}-{completion,shell-hook}`.

**Comment by MdeLv** [comment]
opam init -a (or --auto-setup) does the job.
Tested successfully with two debian 10 machines.
`opam pin --help`...
I don't exactly know what could disable that. Yes, that's certainly a weird behavior related to the exotic _bash_ (should be retired!).

Do you think that a non bash opam tool could workaround that repetitive situation by strictly controlling the environment?
"opamide, opam ide" or else?

**Comment by rjbou** [comment]
If it is on relaunching `opam init --reinit`, it can disable completion, it is fixed in `2.1.0~alpha`. But no other command touches those configuration files, even less pinning.
Opam is compatible with several shells: bash, zsh, {t}csh, and fish. We need to keep those shell interaction to keep terminals and applications (e.g. editors) synced with opam state.

**Comment by MdeLv** [comment]
You should be right.
I may have done opam init --reinit because completion disappeared or seemed to. 
```
       opam-init - Initialize opam state, or set init options.
       --reinit
           Re-run the initial checks and setup, according to opamrc, 
           even if this is not a new opam root
```
btw, is there a detailed opam history stored somewhere?
FYI, I keep detailed history (commands/messages) of all that happens with some crucial "switches".
Thanks.
